test/Vault.spec.ts

```diff
@@ -746,30 +756,36 @@ describe("Vault", () => {
 
     it("transfers underlying from user at sponsor", async () => {
       await vault
-        .connect(alice)
+        .connect(owner)
         .sponsor(underlying.address, parseUnits("400"), TWO_WEEKS);
 
       expect(await vault.totalUnderlying()).to.equal(parseUnits("400"));
       expect(await underlying.balanceOf(vault.address)).to.equal(
         parseUnits("400")
       );
-      expect(await underlying.balanceOf(alice.address)).to.equal(
+      expect(await underlying.balanceOf(owner.address)).to.equal(
         parseUnits("600")
       );
     });
   });
 
   describe("unsponsor", () => {
+    it("reverts if msg.sender is not sponsor", async () => {
```

Comment:
I don't think we need this right? Because you can only unsponsor if you already sponsored and we should still allow people to unsponsor if we remove their role.

Reply:
that makes sense we can remove modifier on unsponsor

Reply:
that is what you mean right? remove only_role for `unsponsor()`?

Reply:
yap

Reply:
and tests :p

Reply:
should be solved. check latest